benchmarks/src/main/java/zipkin2/it/ServerIntegratedBenchmark.java

```diff
@@ -0,0 +1,302 @@
+import zipkin.server.ZipkinServer;
+
+class ServerIntegratedBenchmark {
+
+  static final Logger LOGGER = LoggerFactory.getLogger(ServerIntegratedBenchmark.class);
+
+  static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  @Nullable static final String RELEASED_ZIPKIN_VERSION = System.getenv("ZIPKIN_VERSION");
+
+  static final boolean WAIT_AFTER_BENCHMARK = "true".equals(System.getenv("ZIPKIN_BENCHMARK_WAIT"));
+
+  Closer closer;
+
+  @BeforeEach void setUp() {
+    closer = Closer.create();
+  }
+
+  @AfterEach void tearDown() throws Exception {
+    closer.close();
+  }
+
+  @Test void inMemory() throws Exception {
+    runBenchmark(null);
+  }
+
+  @Test void elasticsearch() throws Exception {
+    GenericContainer<?> elasticsearch = new GenericContainer<>("openzipkin/zipkin-elasticsearch7")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("elasticsearch")
+      .withLabel("name", "elasticsearch")
+      .withExposedPorts(9200)
+      .waitingFor(new HttpWaitStrategy().forPath("/_cluster/health"));
+    startContainer(elasticsearch);
+
+    runBenchmark(elasticsearch);
+  }
+
+  @Test void cassandra() throws Exception {
+    GenericContainer<?> cassandra = new GenericContainer<>("openzipkin/zipkin-cassandra")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("cassandra3")
+      .withLabel("name", "cassandra3")
+      .withExposedPorts(9042)
+      .waitingFor(new LogMessageWaitStrategy().withRegEx(".*Starting listening for CQL clients.*"));
+    startContainer(cassandra);
+
+    runBenchmark(cassandra);
+  }
+
+  @Test void mysql() throws Exception {
+    GenericContainer<?> mysql = new GenericContainer<>("openzipkin/zipkin-mysql")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("mysql")
+      .withLabel("name", "mysql")
+      .withExposedPorts(3306);
+    startContainer(mysql);
+
+    runBenchmark(mysql);
+  }
+
+  void runBenchmark(@Nullable GenericContainer<?> storage) throws Exception {
+    GenericContainer<?> zipkin = startZipkin(storage);
+
+    GenericContainer<?> backend = new GenericContainer<>("openzipkin/example-sleuth-webmvc")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("backend")
+      .withCommand("backend")
+      .withExposedPorts(9000)
+      .waitingFor(new HttpWaitStrategy().forPath("/actuator/health"));
+
+    GenericContainer<?> frontend = new GenericContainer<>("openzipkin/example-sleuth-webmvc")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("frontend")
+      .withCommand("frontend")
+      .withExposedPorts(8081)
+      .waitingFor(new HttpWaitStrategy().forPath("/actuator/health"));
+
+    if (RELEASED_ZIPKIN_VERSION == null) {
+      Testcontainers.exposeHostPorts(9411);
+      String zipkinArg = "-Dspring.zipkin.baseUrl=http://host.testcontainers.internal:9411";
+      backend.withEnv("JAVA_OPTS", zipkinArg);
+      frontend.withEnv("JAVA_OPTS", zipkinArg);
+    }
+
+    startContainer(backend);
```

Comment:
FYI in Testcontainers 1.12.x, there is a helper method to start containers in parallel (including their transitive dependencies):
```
Startables.deepStart(Stream.of(container1, container2)).join()
```

Reply:
Thanks for this tip - the containers start up way faster thanks to this :D